### PR TITLE
Use double instead of float in UCTE model

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
Floats are used everywhere in UCTE network model.

**What is the new behavior (if this is a feature change)?**
Doubles are used in UCTE network model like in IIDM

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by annetill — COMMENTED** [review]
It remains float through the extension `MergedXnode`, do you also need to change that extension. I am going to change the names of some functions in the tests because they contain `Float`. And also, do we need to change also:
```
        if (twoWindingsTransformer.hasProperty(NOMINAL_POWER_KEY)) {
            nominalPower = Float.parseFloat(twoWindingsTransformer.getProperty(NOMINAL_POWER_KEY, null));
        }
```

File: entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnode.java [hunk 1]
```diff
@@ -47,19 +47,19 @@ default MergedXnode setLine1Fictitious(boolean line1Fictitious) {
...
     }
 
-    default MergedXnode setB1dp(float b1dp) {
+    default MergedXnode setB1dp(double b1dp) {
         return this;
     }
 
-    default float getG1dp() {
+    default double getG1dp() {
         return 0.5f;
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnode.java]: Done

File: entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnodeImpl.java [hunk 2]
```diff
@@ -73,8 +73,8 @@ public MergedXnodeImpl(Line line, float rdp, float xdp,
         this.g2dp = checkDividerPosition(g2dp);
     }
 
-    private float checkDividerPosition(float dp) {
-        if (Float.isNaN(dp)) {
+    private double checkDividerPosition(double dp) {
+        if (Double.isNaN(dp)) {
             return 0.5f;
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnodeImpl.java]: Done

File: entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnodeImpl.java [hunk 3]
```diff
@@ -73,8 +73,8 @@ public MergedXnodeImpl(Line line, float rdp, float xdp,
...
         }
         if (dp < 0f || dp > 1f) {
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply entsoe-util/src/main/java/com/powsybl/entsoe/util/MergedXnodeImpl.java]: Done

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java [hunk 4]
```diff
@@ -558,19 +558,19 @@ private static void createPhaseTapChanger(UcteAngleRegulation ucteAngleRegulatio
                 .setRegulationMode(PhaseTapChanger.RegulationMode.FIXED_TAP);
 
         for (int i = lowerTap; i <= Math.abs(lowerTap); i++) {
-            float rho;
-            float alpha;
+            double rho;
+            double alpha;
             double dx = i * ucteAngleRegulation.getDu() / 100f * Math.cos(Math.toRadians(ucteAngleRegulation.getTheta()));
             double dy = i * ucteAngleRegulation.getDu() / 100f * Math.sin(Math.toRadians(ucteAngleRegulation.getTheta()));
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: Done

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java [hunk 5]
```diff
@@ -558,19 +558,19 @@ private static void createPhaseTapChanger(UcteAngleRegulation ucteAngleRegulatio
...
-                    alpha = (float) Math.toDegrees(Math.atan2(dy, 1 + dx));
+                    rho = 1d / Math.hypot(dy, 1d + dx);
+                    alpha = Math.toDegrees(Math.atan2(dy, 1 + dx));
                     break;
 
                 case SYMM:
-                    rho = 1f;
-                    alpha = (float) Math.toDegrees(2 * Math.atan2(dy, 2f * (1 + dx)));
+                    rho = 1d;
+                    alpha = Math.toDegrees(2 * Math.atan2(dy, 2f * (1d + dx)));
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: Done

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java [hunk 6]
```diff
@@ -190,19 +190,19 @@ private static void convertBus(UcteNetwork ucteNetwork, Bus bus, UcteExporterCon
                 geographicalName,
                 getStatus(bus),
                 UcteNodeTypeCode.PQ,
-                Float.NaN,
+                Double.NaN,
                 0.0f,
                 0.0f,
                 0.0f,
                 0.0f,
```
Comment (MioRtia) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java:197]: Should these be floats? If not, you can delete the `f`

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java [hunk 7]
```diff
@@ -380,19 +380,19 @@ private static UcteNode convertXNode(UcteNetwork ucteNetwork, UcteNodeCode xnode
                 geographicalName,
                 ucteNodeStatus,
                 UcteNodeTypeCode.PQ,
-                Float.NaN,
+                Double.NaN,
                 0.0f,
                 0.0f,
                 0.0f,
                 0.0f,
```
Comment (MioRtia) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java:387]: Same remark

**Review by MioRtia — APPROVED** [review]

**Pull request merged**